### PR TITLE
Fix desktop standalone scenario

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/dotnet/cli/issues/2650

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2684)

### CODE REVIEW

File: test/dotnet-publish.Tests/PublishDesktopTests.cs [hunk 1]
```diff
@@ -123,6 +123,25 @@ public class PublishDesktopTests : TestBase
             }
         }
 
+
+        [WindowsOnlyFact]
+        public async Task DesktopApp_WithRuntimes_PublishedSplitPackageAssets()
+        {
+            var testInstance = TestAssetsManager.CreateTestInstance(Path.Combine("..", "DesktopTestProjects", "DesktopAppWithRuntimes"))
```
Comment (piotrpMSFT) [inline test/dotnet-publish.Tests/PublishDesktopTests.cs]: there must be a better approach here than `..`...

Comment (Sridhar-MS) [inline test/dotnet-publish.Tests/PublishDesktopTests.cs]: One option is to create your own `TestAssetsManager` which points to `DesktopTestProjects`.

File: test/dotnet-publish.Tests/PublishDesktopTests.cs [hunk 2]
```diff
@@ -123,6 +123,25 @@ public class PublishDesktopTests : TestBase
...
+                .WithLockFiles();
+
+            var publishCommand = new PublishCommand(testInstance.TestRoot, runtime: "win7-x64");
```
Comment (Sridhar-MS) [inline test/dotnet-publish.Tests/PublishDesktopTests.cs]: Why are you hardcoding runtimes? What about `win7-x86`?

Comment (pakrym) [inline test/dotnet-publish.Tests/PublishDesktopTests.cs]: Single runtime should be enough to run this test.

File: src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs [hunk 3]
```diff
@@ -191,17 +165,8 @@ public ProjectContext Build()
...
-            {
-                // Use the default static resolver
-                frameworkReferenceResolver = FrameworkReferenceResolver.Default;
-            }
-            else
-            {
-                frameworkReferenceResolver = new FrameworkReferenceResolver(ReferenceAssembliesPath);
-            }
+            ReferenceAssembliesPath = ReferenceAssembliesPath ?? FrameworkReferenceResolver.GetDefaultReferenceAssembliesPath();
+            var frameworkReferenceResolver = new FrameworkReferenceResolver(ReferenceAssembliesPath);
```
Comment (anurse) [inline src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs]: Bad Merge? We want to share framework resolver instances as much as possible to reduce I/O reading RedistList files

Comment (pakrym) [inline src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs]: Yep, bad merge, fixing.

**Comment by anurse** [comment]
Let the hacks continue! :shipit :)

**Comment by pakrym** [comment]
@dotnet-bot test OSX x64 Release Build  please

**Pull request merged**